Mathlib/Analysis/Convolution.lean

```diff
@@ -0,0 +1,1681 @@
+  split_ifs with h
+  · rw [intervalIntegral.integral_of_le (le_of_lt h), integral_Ioc_eq_integral_Ioo, ←
+      integral_indicator (measurableSet_Ioo : MeasurableSet (Ioo 0 x))]
+    congr 1 with t : 1
+    have : t ≤ 0 ∨ t ∈ Ioo 0 x ∨ x ≤ t := by
+      rcases le_or_lt t 0 with (h | h)
+      · exact Or.inl h
+      · rcases lt_or_le t x with (h' | h')
+        exacts [Or.inr (Or.inl ⟨h, h'⟩), Or.inr (Or.inr h')]
+    rcases this with (ht | ht | ht)
+    · -- Porting note: was
+      -- rw [indicator_of_not_mem (not_mem_Ioo_of_le ht), indicator_of_not_mem (not_mem_Ioi.mpr ht),
+      --   ContinuousLinearMap.map_zero, ContinuousLinearMap.zero_apply]
+      rw [indicator_of_not_mem (not_mem_Ioo_of_le ht), if_neg (not_mem_Ioi.mpr ht),
+        ContinuousLinearMap.map_zero, ContinuousLinearMap.zero_apply]
+    · -- Porting note: was
+      -- rw [indicator_of_mem ht, indicator_of_mem (mem_Ioi.mpr ht.1),
+      --     indicator_of_mem (mem_Ioi.mpr <| sub_pos.mpr ht.2)]
+      rw [indicator_of_mem ht, if_pos (mem_Ioi.mpr ht.1),
+        if_pos (mem_Ioi.mpr <| sub_pos.mpr ht.2)]
+    · -- Porting note: was
+      -- rw [indicator_of_not_mem (not_mem_Ioo_of_ge ht),
+      --     indicator_of_not_mem (not_mem_Ioi.mpr (sub_nonpos_of_le ht)),
+      --     ContinuousLinearMap.map_zero]
+      rw [indicator_of_not_mem (not_mem_Ioo_of_ge ht),
+        if_neg (not_mem_Ioi.mpr (sub_nonpos_of_le ht)), ContinuousLinearMap.map_zero]
+  · convert (integral_zero ℝ F).symm with t
+    by_cases ht : 0 < t
+    · -- Porting note: was
+      -- rw [indicator_of_not_mem (_ : x - t ∉ Ioi 0), ContinuousLinearMap.map_zero]
+      rw [if_neg (_ : x - t ∉ Ioi 0), ContinuousLinearMap.map_zero]
+      rw [not_mem_Ioi] at h ⊢
+      exact sub_nonpos.mpr (h.trans ht.le)
+    · -- Porting note: was
+      -- rw [indicator_of_not_mem (mem_Ioi.not.mpr ht), ContinuousLinearMap.map_zero,
+      --  ContinuousLinearMap.zero_apply]
+      rw [if_neg (mem_Ioi.not.mpr ht), ContinuousLinearMap.map_zero,
+        ContinuousLinearMap.zero_apply]
+#align pos_convolution_eq_convolution_indicator posConvolution_eq_convolution_indicator
+
+theorem integrable_posConvolution {f : ℝ → E} {g : ℝ → E'} {μ ν : MeasureTheory.Measure ℝ}
+    [SigmaFinite μ]
+    [SigmaFinite ν] [IsAddRightInvariant μ] [NoAtoms ν] (hf : IntegrableOn f (Ioi 0) ν)
+    (hg : IntegrableOn g (Ioi 0) μ) (L : E →L[ℝ] E' →L[ℝ] F) :
+    Integrable (posConvolution f g L ν) μ := by
+  rw [← integrable_indicator_iff (measurableSet_Ioi : MeasurableSet (Ioi (0 : ℝ)))] at hf hg
+  rw [posConvolution_eq_convolution_indicator f g L ν]
+  exact (hf.convolution_integrand L hg).integral_prod_left
+#align integrable_pos_convolution integrable_posConvolution
+
+/-- The integral over `Ioi 0` of a forward convolution of two functions is equal to the product
+of their integrals over this set. (Compare `integral_convolution` for the two-sided convolution.) -/
+theorem integral_pos_convolution [CompleteSpace E] [CompleteSpace E']
```

Comment:
```suggestion
theorem integral_posConvolution [CompleteSpace E] [CompleteSpace E']
```
(I know the statement doesn't actually mention `posConvolution`, though it should).